plugins/woocommerce/includes/admin/reports/class-wc-report-taxes-by-code.php

```diff
@@ -158,18 +158,25 @@ public function get_main_chart() {
 
 		// Merge.
 		$tax_rows = array();
+		// Initialize an associative array to store unique post_ids.
+		$unique_post_ids = array();
 
 		foreach ( $tax_rows_orders + $tax_rows_partial_refunds as $tax_row ) {
-			$key                                    = $tax_row->rate_id;
+			$key                                    = $tax_row->tax_rate;
 			$tax_rows[ $key ]                       = isset( $tax_rows[ $key ] ) ? $tax_rows[ $key ] : (object) array(
 				'tax_amount'          => 0,
 				'shipping_tax_amount' => 0,
 				'total_orders'        => 0,
 			);
-			$tax_rows[ $key ]->total_orders        += 1;
+			$unique_post_ids[ $tax_row->post_id ]   = true;
 			$tax_rows[ $key ]->tax_rate             = $tax_row->tax_rate;
 			$tax_rows[ $key ]->tax_amount          += wc_round_tax_total( $tax_row->tax_amount );
 			$tax_rows[ $key ]->shipping_tax_amount += wc_round_tax_total( $tax_row->shipping_tax_amount );
+			if ( ! isset( $unique_post_ids[ $key ] ) || ! in_array( $tax_row->post_id, $unique_post_ids[ $key ], true ) ) {
+				$unique_post_ids[ $key ]         = isset( $unique_post_ids[ $key ] ) ? $unique_post_ids[ $key ] : array();
+				$unique_post_ids[ $key ][]       = $tax_row->post_id;
+				$tax_rows[ $key ]->total_orders += 1;
+			}
 		}
 
 		foreach ( $tax_rows_full_refunds as $tax_row ) {
```

Comment:
Updating the `$key` below to `tax_rate` as well, should help here, otherwise it may not negate correctly.